File: spring-boot-app/src/main/kotlin/org/example/angulardemo/entity/Job.kt
package org.example.angulardemo.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDate


@Table(name = "job")
data class Job(
    @Id
    val id: String?,
    val userId: Long,
    var title: String,
    var description: String?,
    var startDate: LocalDate,
    var endDate: LocalDate?,
    @Column("current_journal_template_id")
    var currentJournalTemplateId: String?,
)

File: frontend/src/app/shared/job.ts
export class Job {
  id?: string;
  userId: number;
  title: string;
  description: string;
  startDate: string;
  endDate?: string;
  currentJournalTemplateId?: string;

  constructor(userId: number, title: string, description: string, startDate: string, id?: string, currentJournalTemplateId?: string) {
    this.id = id;
    this.userId = userId;
    this.title = title;
    this.description = description;
    this.startDate = startDate;
    this.currentJournalTemplateId = currentJournalTemplateId;
  }
}
